### PR TITLE
Add database migration project

### PR DESCRIPTION
**Pull request by enchobelezirev**
#### Description: 
PostgreSQL migration tool - PoC

#### Issue: 
https://sapjira.wdf.sap.corp/browse/LMCROSSITXSADEPLOY-1992

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseMigrationExecutor.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public abstract DataSource getSourceDataSource();
+
+    public abstract DataSource getTargetDataSource();
+
+    public void executeMigration(String entityName) {
+        try {
+            executeMigrationInternal(entityName);
+        } catch (SQLException e) {
+            LOGGER.error(MessageFormat.format("Error migrating entity \"{0}\"", entityName), e);
+            throw new IllegalStateException(e.getMessage(), e);
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseMigrationExecutor.java:25]: `IllegalStateException`s should be used only for things that should not happen. It's foreseeable for an error to occur here, so we should use some other exception type. Also, we should include `"Error migrating entity \"{0}\": {1}"` in the exception message, where 1 is the exception message of the cause.

File: com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/generator/DatabaseTableInsertQueryGenerator.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String CLOSING_BRACKET = ")";
+    private static final String OPEN_BRACKET = "(";
+    private static final String DEFAULT_STATEMENT_VALUES_SEPARATOR = ", ";
+    private static final String DEFAULT_STATEMENT_PARAMETER = "?";
+
+    public String generate(DatabaseTableMetadata tableMetadata) {
+        return generateInsertStatement(tableMetadata);
+    }
+
+    private String generateInsertStatement(DatabaseTableMetadata tableMetadata) {
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/generator/DatabaseTableInsertQueryGenerator.java:21]: Why not inline this method in the one above and rename it to `generateInsertStatement`?

File: com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/extractor/DataSourceEnvironmentExtractor.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    public DataSource extractDataSource(String serviceName) {
+        LOGGER.info(MessageFormat.format("Extracting datasource for service {0}...", serviceName));
+        CfCredentials databaseServiceCredentials = extractDatabaseServiceCredentials(serviceName);
+        return extractDataSource(databaseServiceCredentials);
+    }
+
+    private CfCredentials extractDatabaseServiceCredentials(String serviceName) {
+        CfEnv cfEnv = new CfEnv();
+        CfService sourceDatabase = cfEnv.findServiceByName(serviceName);
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/extractor/DataSourceEnvironmentExtractor.java:26]: Rename this variable to `service`.

File: com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseTableMigrationExecutor.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                                                                                             .build());
+        }
+        return tableMetadataBuilder.build();
+    }
+
+    private void transferData(String databaseTable, DatabaseTableMetadata sourceTableMetadata) throws SQLException {
+        LOGGER.info(MessageFormat.format("Transfering data for table \"{0}\"...", databaseTable));
+        String insertQuery = new DatabaseTableInsertQueryGenerator().generate(sourceTableMetadata);
+        getSqlQueryExecutor(getSourceDataSource()).executeWithAutoCommit((sourceConnection) -> {
+            getSqlQueryExecutor(getTargetDataSource()).executeWithAutoCommit((targetConnection) -> {
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseTableMigrationExecutor.java:65]: Will this create a transaction. Doesn't `autoCommit` mean that each statement will be in its own transaction?

File: com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseTableMigrationExecutor.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                                                                                             .columnType(resultSetMetadata.getColumnTypeName(currentColumnIndex))
+                                                                                             .build());
+        }
+        return tableMetadataBuilder.build();
+    }
+
+    private void transferData(String databaseTable, DatabaseTableMetadata sourceTableMetadata) throws SQLException {
+        LOGGER.info(MessageFormat.format("Transfering data for table \"{0}\"...", databaseTable));
+        String insertQuery = new DatabaseTableInsertQueryGenerator().generate(sourceTableMetadata);
+        getSqlQueryExecutor(getSourceDataSource()).executeWithAutoCommit((sourceConnection) -> {
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseTableMigrationExecutor.java:64]: Do we need a transaction for the source database? We're just doing one SELECT to it.

**Comment by enchobelezirev** [comment]
Unit tests must be implemented.

**Pull request merged**